### PR TITLE
feat: make Tables a Mapping

### PR DESCRIPTION
**Pull request by NickCrews**
Fixes https://github.com/ibis-project/ibis/issues/8672

### CODE REVIEW

File: ibis/expr/types/relations.py [hunk 1]
```diff
@@ -516,6 +516,24 @@ def preview(
...
+
+        Returns
+        -------
+        list
+            Column names
+
+        Examples
+        --------
+        >>> import ibis
+        >>> ibis.options.interactive = True
```
Comment (cpcloud) [inline ibis/expr/types/relations.py:530]:
```suggestion
```

This isn't necessary, you're not displaying any Ibis expressions.

**Review by cpcloud — CHANGES_REQUESTED** [review]

File: ibis/tests/expr/test_table.py [hunk 2]
```diff
@@ -123,6 +123,19 @@ def test_table_tab_completion():
     assert items.issuperset(table.columns)
 
 
+def test_keys(table):
+    keys = table.keys()
+    assert keys == table.columns
+
+
```
Comment (cpcloud) [inline ibis/tests/expr/test_table.py:130]:
```suggestion
```

The test below seems sufficient. Alternatively you can put this assertion there.

**Review by cpcloud — CHANGES_REQUESTED** [review]

**Comment by cpcloud** [comment]
Are you planning on making `Table` a proper `Mapping` subclass, or are you just interested in the kwarg unpacking behavior?

**Comment by jcrist** [comment]
If you're only interested in kwarg unpacking in `ibis.struct`, could we instead just make `ibis.struct(table)` work (adding a special case for a `Table` arg) rather than relying implicitly on kwarg unpacking? This will be both more efficient (since we won't need to do the work to validate all columns share a parent relation) and will let us avoid the addition of a `.keys()` method on a table (which IMO feels pretty weird).

FWIW my thoughts [here](https://github.com/ibis-project/ibis/pull/8671#issuecomment-2007967971) on making structs a mapping apply equally to the case here.

**Comment by NickCrews** [comment]
I talk about most of these concerns in https://github.com/ibis-project/ibis/issues/8672.

> only interested in kwarg unpacking in ibis.struct

Not just that, I want to take advantage of generic ** unpacking to do `{"col1": "default_that_might_be_overwritten", **table, "col2": "always_this_val"}`.

**Comment by NickCrews** [comment]
closing for now as stale.

**Pull request closed without merging**